pyccel/codegen/printing/cwrappercode.py

```diff
@@ -110,21 +141,25 @@ def _print_FunctionDef(self, expr):
         keyword_list_name = self.get_new_name(used_names,'kwlist')
         keyword_list = PyArgKeywords(keyword_list_name, arg_names)
 
+        code = ''
         wrapper_body = [keyword_list]
         wrapper_body_translations = []
 
         parse_args = []
         # TODO: Simplify (to 1 line?)
         # TODO: Handle optional args
         for a in expr.arguments:
-            collect_type, cast_func = self.get_PyArgParseType(a.dtype)
+            collect_var, cast_func = self.get_PyArgParseType(used_names, a)
             if cast_func is not None:
                 # TODO: Add other properties
-                collect_var = Variable(dtype=collect_type,
-                        name=self.get_new_name(used_names, a.name+"_tmp"))
+                cast_func = self.pop_cast_function(cast_func)        
+                if not cast_func in self._cast_functions_set:
+                    code += self._print(cast_func)
```

Comment:
This prints the definition of the function inside the current function. Are you sure that is what we want? If so what is the point in `self._cast_functions_set`?

Reply:
Ah! No it is printed before the function. However this will lead to a not very neat code:
```
#include <Python.h>

_Bool pyint_to_bool(...) {...}

PyObject* user_func(...) {...}

_Complex pyobject_to_complex(...) {...}

PyObject* user_func2(...) {...}

PyObject* user_func3(...) {...}

module
```

It would be better to group the casting functions together.

This can be done by printing them in `_print_Module`